pkg/reconciler/taskrun/validate_resources.go

```diff
@@ -78,6 +78,11 @@ func validateParams(ctx context.Context, paramSpecs []v1beta1.ParamSpec, params
 	if extraParamsNames := extraParamsNames(ctx, neededParamsNames, providedParamsNames); len(extraParamsNames) != 0 {
 		return fmt.Errorf("didn't need these params but they were provided anyway: %s", extraParamsNames)
 	}
+	// This is needed to support array replacements in params.Users want to use $(results.resultname[*])
+	// to pass array result to array param, yet in yaml format this will be
+	// unmarshalled to string for ArrayOrString. So we need to check and correct the param type.
+	// Please check issue #4879 for more details and examples.
+	correctAoStype(params, neededParamsTypes)
```

Comment:
This actually worths adding a separate test in validate_resource_test for checking the changed params

Reply:
sure! That's a good point!